cogeo_mosaic/backends/file.py

```diff
@@ -0,0 +1,56 @@
+import functools
+import json
+from typing import Dict, Optional, Tuple, Union
+
+import mercantile
+
+from cogeo_mosaic.backends.base import BaseBackend
+from cogeo_mosaic.backends.utils import (
+    _compress_gz_json,
+    _decompress_gz,
+    get_assets_from_json,
+)
+from cogeo_mosaic.model import MosaicJSON
+
+
+class FileBackend(BaseBackend):
+    """Local File Backend Adapter"""
+
+    def __init__(
+        self, path: str, mosaic_def: Optional[Union[MosaicJSON, Dict]] = None,
+    ):
+        self.path = path
+        if mosaic_def is not None:
+            self.mosaic_def = MosaicJSON(**dict(mosaic_def))
+        else:
+            self.mosaic_def = self.read_mosaic(path)
+
+    def tile(self, x: int, y: int, z: int) -> Tuple[str]:
+        """Retrieve assets for tile."""
+
+        return get_assets_from_json(self.mosaic_def.tiles, self.quadkey_zoom, x, y, z)
+
+    def point(self, lng: float, lat: float) -> Tuple[str]:
+        """Retrieve assets for point."""
+        tile = mercantile.tile(lng, lat, self.quadkey_zoom)
+        return get_assets_from_json(
+            self.mosaic_def.tiles, self.quadkey_zoom, tile.x, tile.y, tile.z
+        )
+
+    def upload(self):
+        with open(self.path, "wb") as f:
+            f.write(_compress_gz_json(dict(self.mosaic_def)))
```

Comment:
compression could be an option ?

Reply:
Good point. Also when compression is used should a `.gz` suffix be added to the filename? Both here and with S3?

Reply:
I think the `.gz` is already in the filename -> so we apply compression

but I'm not 💯 sure this is the best way 🤔

Reply:
Maybe
```
def upload(self, gzip=None):
```
if `gzip` is `True`, apply compression, if `False`, don't, and if `None` apply compression if `url`/`key`/`path` ends in `.gz`

Reply:
sounds good

Reply:
Updated to be smarter with compression as described above.